### ISSUE TITLE
Install Script Fails When Behind Proxy

### ISSUE THREAD

**Issue by edcs** [getsentry/sentry-cli#228]
The `bin/install-sentry-cli` script uses Node's HTTP and HTTPS libraries to download the Sentry CLI binary. Node.js doesn't have a way of globally applying proxy settings meaning that you need to modify the install script to apply them - this isn't very practical because there's probably lots of people who aren't trapped behind a corporate proxy.

Would it be feasible to change this script to use `curl` if available to do the download? Or is there another way to grab the file? I'm using Sentry in a Rect Native project and trying to install using Yarn.

**Comment by edcs** [comment]
Or as an alternative, could the script skip installation if the Sentry CLI has already been installed via Homebrew or another alternative method?

**Comment by jan-auer** [comment]
Hi @edcs and thanks for bringing this up! To be fair, our installation script has been pretty rudimental so far and I guess it's time to change this now.

Proxy: Happy to accept a PR for this. However, it would be good to find a solution that ideally works the same on Windows, Linux and macOS. After reading up on this a little bit, you must be having the same issue with `npm` itself as well. Do people behind proxies usually configure npm's proxy setting? If so, we could leverage this, migrate to [node-fetch](https://npmjs.org/package/node-fetch) and add the proxy in the download script like [here](https://github.com/bitinn/node-fetch/issues/79#issuecomment-184594701). That way, all systems are supported, even Windows and those without curl.

Homebrew: This is a good idea, but I'm looping in @HazAT here for a second opinion. In general, I'd be fine if you **copy** the exectuable to the right location iff the version matches exactly. Even though we try to not break between major releasees, we had to make some adaptions in the past and do not test all version combinations when we push out a release. From my side, PR for this is more than welcome anytime.

**Comment by edcs** [comment]
Hi @jan-auer - To get Yarn or NPM to work behind a proxy, you set the config options you linked to and they just work. Unfortunately Node has no such setting.

You can set a proxy agent in `http` and `https`, `node-fetch` seems to use the same configuration. I did experiment with this idea yesterday but couldn't get it to work, hence why I thought CuRL would be a good alternative. However, if I can get Node configured properly that's possibly the best option. My idea was to include the agent if `HTTP_PROXY`/`HTTPS_PROXY` were defined in the environment.

**Comment by jan-auer** [comment]
@edcs NPM injects each of its config values via environment variable in all scripts that run during the lifecycle. This includes the install script that downloads sentry-cli. So by reading `process.env.npm_config_https_proxy` (or similar) it should be possible to get that value and then reconfigure the download. 

Also supporting `process.env.HTTPS_PROXY` is a good idea, but it would seem that the npm config should have been set anyways. So probably something like:

```js
var proxy = process.env.npm_config_https_proxy || process.env.HTTPS_PROXY;
```

Please excuse, that I don't have the time to change and test this myself right now.

**Comment by jan-auer** [comment]
Released with `1.29.0`. Thanks for your input and the PR 🎉
